packages/eslint-plugin/docs/rules/ban-types.mdx

```diff
@@ -15,6 +15,16 @@ It's often a good idea to ban certain types to help with consistency and safety.
 This rule bans specific types and can suggest alternatives.
 Note that it does not ban the corresponding runtime objects from being used.
 
+:::danger Deprecated
+This rule is deprecated and now split across several rules:
```

Comment:
Why are we deprecating the rule?
There's still value in the rule - just that the default config can be empty now.

People use this rule all the time to ban various types across their codebase. It's the only way to reliably target type names only without hitting value names.

Reply:
@bradzacher because it's essentially being renamed to `no-restricted-types`, with the default values split into more targeted rules. I'll update the deprecation notice to make it more clear.

Reply:
Given this is a major - should we just remove the old rule name entirely?
Is there a benefit to keeping it if we're adding the new rules to recommended and the old rule should almost be a 1:1 name swap in config?

Reply:
I was thinking this is a pretty user-known rule so leaving it in would help with migration. The name swap might be 1:1 but the behavioral means figuring out which of the three new rules individual config entries / inline comments map to.

Reply:
Makes sense to me too...
If we do that for this rule, should we just go ahead and do that for https://github.com/typescript-eslint/typescript-eslint/pull/8821, too, though?

Reply:
@kirkwaiblinger we can land #8821 against main and then remove it in v8
similarly we could technically land this PR in main and then remove `ban-types` in v8 (which is what I originally suggested https://github.com/typescript-eslint/typescript-eslint/pull/8977#issuecomment-2079376537).

I.e. we could backport the new rules to v7 so they're available right now, deprecate `ban-types`, and then do the actual breaking change (removing `ban-types`) in v8.

Reply:
I like it. Once this PR goes in I can send a followup to backport the changes to v7 / `main`, replacing the deletion with a deprecation.